packages/langium/src/test/langium-test.ts

```diff
@@ -0,0 +1,123 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { CompletionItem, DocumentSymbol, Range } from 'vscode-languageserver';
+import { LangiumDocument } from '../documents/document';
+import { LangiumServices } from '../services';
+import { getDocument } from '../utils/ast-util';
+
+export interface LangiumTest {
+    expectSymbols(input: string, ...expectedItems: Array<string | Partial<DocumentSymbol>>): void
+    expectCompletion(input: string, index: number, ...expectedItems: Array<string | Partial<CompletionItem>>): void
+    expectGoToDefinition(input: string, index: number, rangeExpectation: number): void
+}
+
+export type ExpectFunction = (actual: unknown, expected: unknown) => void;
+
+export function expectSymbols(services: LangiumServices, cb: ExpectFunction): (input: string, ...expectedSymbols: DocumentSymbol[]) => void {
+    return (input, ...expectedSymbols) => {
+        const document = parseDocument(services, input);
+        if (!document.parseResult) {
+            throw new Error('Could not parse document');
+        }
+        const symbolProvider = services.lsp.DocumentSymbolProvider;
+        const symbols = symbolProvider.getSymbols(document);
+        cb(symbols.length, expectedSymbols.length);
+        for (let i = 0; i < expectedSymbols.length; i++) {
+            const expected = expectedSymbols[i];
+            const item = symbols[i];
+            if (typeof expected === 'string') {
+                cb(item.name, expected);
+            } else {
+                cb(item, expected);
+            }
+        }
+    };
+}
+
+export function expectCompletion(services: LangiumServices, cb: ExpectFunction): (input: string, index: number, ...expectedItems: Array<string | CompletionItem>) => void {
```

Comment:
I'd propose to define the actual test parameters as an object, so it's easier to add optional parameters like changing the `<|>` marker string, and parameters of boolean and number type (like `index`) are more readable on the call site.